packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -291,25 +294,48 @@ export class ChainService implements ChainServiceInterface {
     subscriptions.map(s => s.unsubscribe());
   }
 
+  private async waitForConfirmations(event: Event | null): Promise<void> {
```

Comment:
Seems like a kind of strange API to me. When is `event` going to be `null`?

Reply:
We probably also can just pass in a transaction hash object instead of the entire ethers `Event`

Reply:
Good call on limiting the parameter to be a transaction.

When a wallet registers a channel, the chain service reads the funding amount from the chain and calls `holdingUpdated` with the value. This read is **not** associated with a transaction. However, we still want to protect against chain reorgs for this read.

The goal with emitting a funding value after channel registration is to allow wallets that start after some downtime to have a correct funding value in the database.

Reply:
On second thought, I think it might be cleaner to stick with passing an ethers event into this function. If this functionality needs to be reused, we can do more refactoring/rethinking of function signatures.

Also, I cleaned up the types a bit de5085582